smartsim/_core/entrypoints/colocated.py

```diff
@@ -165,21 +165,27 @@ def main(
 ):
     global DBPID
 
+    lo_address = current_ip("lo")
     try:
-        ip_address = None
-        if network_interface:
-            ip_address = current_ip(network_interface)
-        lo_address = current_ip("lo")
+        ip_addresses = []
+        ip_addresses.extend(
+            [current_ip(interface) for interface in network_interface.split(",")]
+        )
     except ValueError as e:
         logger.warning(e)
-        ip_address = None
+        ip_addresses = []
 
-    if lo_address == ip_address or not ip_address:
+    if (
+        all([lo_address == ip_address for ip_address in ip_addresses])
```

Comment:
minor "performance" thing: If you drop the square brackets here, `all` will evaluate this lazily and will return on the first instance of `False`

That said, this list is small and we are unlikely to see literally any difference, so this is an entirely optional refactor